compose/app/src/main/java/com/chartboost/mediation/sdk/demo/AdController.kt

```diff
@@ -0,0 +1,501 @@
+            }
+
+            AdType.INTERSTITIAL ->
+                if (shouldUseFullscreenApi.value) showFullscreenAd(
+                    context,
+                    logState
+                ) else showInterstitialAd(
+                    logState
+                )
+
+            AdType.REWARDED_VIDEO ->
+                if (shouldUseFullscreenApi.value) showFullscreenAd(
+                    context,
+                    logState
+                ) else showRewardedAd(
+                    logState
+                )
+
+            AdType.REWARDED_INTERSTITIAL -> showFullscreenAd(context, logState)
+        }
+    }
+
+    /**
+     * Loads a banner ad.
+     *
+     * @param context The context.
+     * @param placementName The name of the placement to load.
+     * @param size The size of the banner ad.
+     * @param logState The log state to update.
+     */
+    fun loadBannerAd(
+        context: Context,
+        placementName: String,
+        size: HeliumBannerAd.HeliumBannerSize,
+        logState: MutableList<String>,
+    ): HeliumBannerAd {
+        logState.add("Banner ad is about to load")
+
+        return HeliumBannerAd(context, placementName, size, object : HeliumBannerAdListener {
+            override fun onAdCached(
+                placementName: String,
+                loadId: String,
+                winningBidInfo: Map<String, String>,
+                error: ChartboostMediationAdException?,
+            ) {
+                logState.add("Banner ad loaded ${if (error != null) "with error: ${error.chartboostMediationError}" else "successfully"}")
+            }
+
+            override fun onAdClicked(placementName: String) {
+                logState.add("Banner ad clicked")
+            }
+
+            override fun onAdImpressionRecorded(placementName: String) {
+                logState.add("Banner ad impression recorded")
+            }
+        })
+    }
+
+    /**
+     * Handles the initialization results.
+     *
+     * @param error The error, if any.
+     * @param continuation The continuation to resume.
+     * @param logState The log state to update.
+     */
+    private fun handleInitializationResults(
+        error: Error?,
+        continuation: Continuation<Result<Unit>>,
+        logState: MutableList<String>,
+    ) {
+        // [Optional] Subscribe to partner initialization results to get the initialization metrics data.
+        HeliumSdk.subscribeInitializationResults(object : PartnerInitializationResultsObserver {
+            override fun onPartnerInitializationResultsReady(data: PartnerInitializationResultsData) {
+                logState.add("Partner initialization metrics data is available: ${data.data}")
+            }
+        })
+
+        continuation.resume(if (error != null) Result.failure(error) else Result.success(Unit))
+        logState.add("Chartboost Mediation SDK initialization ${if (error != null) "failed with error: ${error.cause}" else "succeeded"}")
```

Comment:
shouldn't we treat resume as 'return'? or else you could have a bunch of stuff execute between the resume and this line.